README.md

```diff
@@ -2,44 +2,119 @@
+
+***NS records in basedomain***
+
+dns-operator-azure is supposed to create an `NS` record in the `<baseDomain>` zone to refer the `<wc_name>.<base_domain>` zone.
+
+### Public DNS Zone <mc_name>.<base_domain> in <mc_name> resource group
+Since MC is also a workload cluster in itself, this entry is supposed to exist as above.
+ 
+### Private DNS Zone <wc_name>.<base_domain> in <wc_name> resource group 
+- Created only for private CAPZ workload clusters by CAPZ.
+- Used by the WC itself.
+
+***A records***
+- `apiserver` should refer `AzureCluster.Spec.NetworkSpec.APIServerLB.FrontendIPs[0].PrivateIPAddress`
+- `ingress` should refer the IP of k8s service in the WC if exists. Managed by `external-dns` (deployed by default-apps-azure)
+ 
+### Private DNS Zone <wc_name>.<base_domain> in <mc_name> resource group 
+
+- Used by the MC to access the WC api.
+- Created for private CAPZ workload clusters by dns-operator-azure.
+- MC can be public or private. It doesn't matter.
+- Not created for non-CAPZ workload clusters.
+
+
+***A records***
+- `apiserver` should refer the private endpoint in the MC VNet that points the private link of WC api.
+
+###  Private DNS Zone <mc_name>.<base_domain> in <mc_name> resource group
+
+Since MC is also a workload cluster in itself, this entry is supposed to exist as above if MC is a private cluster..
+
+###  Private DNS Zone <mc_name>.<base_domain> in <wc_name> resource group  
+
+- Used by the WC to access the MC ingress endpoints.
+- Created for private CAPZ management clusters by dns-operator-azure.
+- WC can be public or private. It doesn't matter.
+- Not created for non-CAPZ workload clusters.
+- The ingress record refers the private endpoint in the WC VNet that points the private link of MC ingress endpoints.
+
+***A records***
+- `ingress` should refer the private endpoint in the WC VNet that points the private link of MC ingress.
+
+***CNAME records***
+- `*` should refer `ingress.<mc_name>.<base_domain>`. 
+
+## Bastion concept
+
+We switched to `teleport` to access cluster nodes, so we stopped deploying bastion nodes. This feature is no 
+supported anymore. Last supported version was `1.3.4`. 
+
+## Cluster Deletion
+
+On `Cluster` deletion, `CAPZ` deletes the entire `resourceGroup` where the `<clustername>` specific DNS zone exists 
+as well. For that reason on deletion only the `NS` record in the `<baseDomain>` must be handled by the operator.
+
+
+## Configuration of the operator
 
 `dns-operator-azure` expect an existing DNS Zone which is used as `baseDomain` (e.g. `kubernetes.my-company.io`).
 There is no need to have this DNS Zone in the same subscription where `dns-operator-azure` is running.
 
 To act on this DNS Zone, the name and the resource group must be defined by `-base-domain` and `-base-domain-resource-group` flag.
 The subscription where this DNS Zone exist must be defined by setting the `AZURE_SUBSCRIPTION_ID` environment variable.
 
-### Azure AuthN/AuthZ
+## Azure AuthN/AuthZ
```

Comment:
@vvondruska Can you describe the authentication mechanism you introduced here?

Reply:
There are no changes in this authentication mechanism.
I just added a new one to be used for the non-Azure clusters because the WCs may need to be placed in a different subscription than the base domain, and in that case they require their own authentication.
There are 2 ways to provide Azure authentication for non-Azure WCs:
1. Use Azure credentials from MC
2. Provide credentials for non-Azure WCs in additional flags and env variables.

I will try to describe it in the README file.

Reply:
Thank you so much! May I get a review now? 🙏